### PR TITLE
Adding package.json to add npm install support.

### PR DESCRIPTION
**Pull request by jamesleesaunders**
Towards #732. Add package.json to enable x3dom to be installed with npm.

### CODE REVIEW

**Comment by jamesleesaunders** [comment]
As hinted at in issue #732:
> I would suspect that adding automatic publishing to npm (perhaps first to an external account) to the (python) build system would be probably a welcome contribution as long as it does not add any manual work load.
I will next look to see if we can add automatic publishing to npm as part of the python build process.
Would need to contact the current owner @fibo of the https://www.npmjs.com/package/x3dom package to see if he is willing to add other collaborators access to upload new releases to npm (I am happy to volunteer if needed). It would be nice to make https://www.npmjs.com/package/x3dom the 'official' npm package rather than start a new one with a different name.

**Comment by jamesleesaunders** [comment]
@fibo : good spot on the ‘jamesleesaunders’ in the url! Oops! Fixed!
Please forgive me if I am stepping on any toes, I was just thinking it would be nice to get the ‘official’ x3dom published on npm.

x3dom devs : would you consider moving the ‘dist’ folder (currently in separate repo) into this main repo, and then use and .npmignore file to exclude them from the npm registr upload?

I am still learning the intricacies of packaging to npm, so please let me know if I am barking up the wrong tree.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	"license": "MIT OR GPL-3.0",
+	"keywords": [
+		"3d",
+		"x3d",
+		"x3dom",
+		"webgl"
+	],
+	"homepage": "http://x3dom.org",
+	"main": "dist/x3dom.js",
+	"style": "dist/x3dom.css",
```
Comment (jamesleesaunders) [inline package.json]: I appreciate the 'dist' folder does not exist in this repo. Would it be possible to move the dist folder from https://github.com/medihack/x3dom-full-dist into this same repo?

**Comment by fibo** [comment]
Also attributes *files* and *issues* should be edited. There is already the dist repo, I think it is ok as is.

**Comment by jamesleesaunders** [comment]
Thanks @fibo if it is preferred to keep the dist in a separate repo then no worries. I will look forward to v1.7.3 being published (on GitHub and npm) at some point in the future.

I would like to contribute to x3dom if there is something simple I can do. I’ll have another look down the issues list to see if there is anything else simple I can help with, or any documentation I can embellish.

All the best. Jim

**Comment by fibo** [comment]
@jamesleesaunders if you want to help maintaining dist repo you are welcome.

It requires a very little effort, just copying the latest x3dom and publish it.

Currently there is an issue in this repo open with the possibility to add also x3dom full to the dist.

I checked that current version is 1.7.3-dev probably stable version will arrive soon.

You can cobtribute with a PR on dist repo or if you prefer I can try to add you (I think I should have permissions) as a collaborator to the dist repo, as you prefer.

**Comment by jamesleesaunders** [comment]
Thank you @fibo I don’t think there is a need to make me a collabotator on the dist repo (just yet), if/when the 1.7.3 time comes I am happy to submit it in via PR. Like you say looks easy but still interested. Thanks for your guidance. All the best, Jim

**Pull request closed without merging**